### DISCUSSION TITLE
Can you change values of certain sources (like Text GDI+) with obsws-python?

### DISCUSSION THREAD

**Discussion by ErezCodes** [aatikturk/obsws-python#69] [Q&A] [answered]
Is there a part in the code that allows you to change the value of a Text or an image?

**Comment by aatikturk** [comment, marked as answer]
Hi,

That's possible using ```ReqClient``` class.

For initial connection,  (from the example in README)

```python
>>>import obsws_python as obs

# pass conn info if not in config.toml
>>>cl = obs.ReqClient(host='localhost', port=4455, password='mystrongpass', timeout=3)
```

Now you have a client to make requests to OBS websocket.
- First you need the settings of your input (for your case, text-input)
Assume name of the text input is ```test_string```.  To get the settings


```python
>>> resp = cl.get_input_settings("test_string")

# to access the settings inside the resp use  input_settings attribute

>>>print(resp.input_settings)

{'text': 'hello world'}
```

Now you know what input settings look like. So you can use this as a template and change the text as you wish  using ```set_input_settings``` method


```python
>>> new_settings = {'text': 'WHATS UP'}
>>> cl.set_input_settings("test_string", new_settings, True)
# now text should be changed.
```


Details for each method and their parameters can be found using docstrings 

```python
>>> help(cl.set_input_settings)
```


Hope it helps
